src/NServiceBus.Core/Audit/IAuditContext.cs

```diff
@@ -22,6 +24,20 @@ public interface IAuditContext : IBehaviorContext
         /// </summary>
         /// <param name="key">The audit key.</param>
         /// <param name="value">The value.</param>
+        [ObsoleteEx(
+            ReplacementTypeOrMember = nameof(AuditMetadata),
+            TreatAsErrorFromVersion = "9.0",
+            RemoveInVersion = "10.0")]
         void AddAuditData(string key, string value);
+
+        /// <summary>
+        /// Metadata for the audited message.
+        /// </summary>
+        IDictionary<string, string> AuditMetadata { get; }
```

Comment:
One downside of using IDictionary is that if you want to ever turn the data into IReadOnlyDictionary somewhere else it is not possible without copying the data. This is one of the reason I switched to Dictionary directly (which is also aligned with other usages of key value pair data exposed on contexts if I recall correctly)

Reply:
Makes sense, fixed now